backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class FindMatchQuery {
+
+  private Date date;
+  
+  /** Constructor */
+  public FindMatchQuery() {
+    date = new Date();
+  }
+
+  /** Constructor with manually set date*/
+  public FindMatchQuery(Date date) {
+    this.date = date;
+  }
+
+  /** Find match of new participant or add to list of participants, return whether or not match was found 
+      right after being added */
+  public Match findMatchQuery(List<Participant> participants) {
+    
+    // Compare new participant preferences with others in list to find match
+    int numParticipants = participants.size();
+    Participant newParticipant = participants.get(numParticipants - 1);
+    for (int i = 0; i < numParticipants - 1; i++) {
+      Participant currParticipant = participants.get(i);
+      
+      // Check if participants are looking for similar meeting duration
+      int newDuration = newParticipant.getDuration();
+      int currDuration = currParticipant.getDuration();
+      boolean compatibleDuration = Math.abs(newDuration - currDuration) <= 15;
```

Comment:
Please extract `15` to a private static final constant with a name that explains its use like "MINIMUM_MEETING_TIME". Doing so helps future readers of your code understand the significance of the value and also makes it easy to change if it's referenced multiple times in the code (as it is here)

Reply:
+1

Reply:
The 15s are actually referring to different things, but I made constants for both for clarity.